### ISSUE TITLE
Cell bookmarking for fast navigation inside a single notebook

### ISSUE THREAD

**Issue by rodsenra** [jupyterlab/jupyterlab#2916]
Use Case:
Whenever we have a very long notebook, to interactively execute groups of cells that are physically far apart each other, the user needs to either scroll up and down the page or use a ToC.

Desired Feature:
Use hotkeys to create *ad hoc* cell bookmarks that allow direct navigation between cells.
The operation could be equivalent to [bookmarking in Emacs for example](https://www.gnu.org/software/emacs/manual/html_node/emacs/Bookmarks.html).

Possible Approaches:
* When creating a bookmark we could set the metadata name cell (if it has not one already defined) and bind a hotkey to jump to a cell with that name. 
* Create a cell metatag (e.g. "__bookmark__") and a hotkey would by default jump over all cell with that tag

**Comment by rodsenra** [comment]
I am open to further discuss the use case and possible solutions (here or other channel), as well as volunteering to work on this on spare time.

**Comment by epinar** [comment]
Indeed, for example when we reload a notebook and execute many cells which are ordered logically whereas having unrelated cells in between, either a bookmark or a "jump from a cell to another cell" would be very helpful to connect the related cells. The second approach would allow executing cells in a mixed manner. 
Approach: Tag the cells by "bookmark-1", "bookmark-2", so on...

**Comment by kevin-j-le** [comment]
i know it's been a while, but has anyone implemented bookmarks yet?

**Comment by TiesdeKok** [comment]
I recently created a JupyterLab extension that mimics the bookmarks extension in VSCode, which provides functionality similar to what is requested in this thread. 😃 You can find it here: https://github.com/TiesdeKok/jupyterlab_code_bookmarks
